### PR TITLE
feat(legacy): confirm ETH stake, unstake, claim

### PR DESCRIPTION
**Pull request by romanz**
Similar to https://github.com/trezor/trezor-firmware/pull/3534.

 - [x] add staking-specific tests (both positive and negative).
 - [x] validate arguments from `data_initial_chunk`.

# New confirmation layouts

### CODE REVIEW

**Comment by romanz** [comment]
ETH device tests succeeded also on a T1B1 device:

```
$ pytest 'tests/device_tests/ethereum/'
<snip>
=================== 157 passed, 26 skipped in 905.91s (0:15:05) ===================
```

**Review by mmilata — COMMENTED** [review]
Just some nits otherwise LGTM

File: legacy/firmware/ethereum.c [hunk 1]
```diff
@@ -567,8 +560,147 @@ static void ethereum_signing_handle_erc20(struct signing_params *params,
...
+// Returns `true` if it is a staking-related transaction and updates `op` with
+// its specific operation.
+static bool isEthereumStakingTx(const struct signing_params *params,
+                                enum staking_operation_t *op) {
+  if (params->data_initial_chunk_size < SC_FUNC_SIG_BYTES) {
+    return false;
+  }
+  bool is_address_pool =
+      ((memcmp(params->pubkeyhash, POOL_HOLESKY_TESTNET, 20) == 0) ||
+       (memcmp(params->pubkeyhash, POOL_MAINNET, 20) == 0));
```
[the remark this answers is not in the document]
Comment (romanz) [reply legacy/firmware/ethereum.c]: https://github.com/trezor/trezor-firmware/pull/4930/commits/0f4ded9298b60fef1492e32ab5e9320e0ea835d5

File: legacy/firmware/ethereum.c [hunk 2]
```diff
@@ -331,14 +331,19 @@ static void ethereumFormatAmount(const bignum256 *amnt,
   bn_format(amnt, NULL, suffix, decimals, 0, false, ',', buf, buflen);
 }
 
+static void parse_bignum256(const uint8_t *value, uint32_t value_len,
+                            bignum256 *result) {
+  uint8_t padded[32] = {0};
+  memzero(padded, sizeof(padded));
```
[the remark this answers is not in the document]
Comment (romanz) [reply legacy/firmware/ethereum.c]: https://github.com/trezor/trezor-firmware/pull/4930/commits/528b899739f2229d00d24f7072977ba597925c74

File: legacy/firmware/ethereum.c [hunk 3]
```diff
@@ -567,8 +560,147 @@ static void ethereum_signing_handle_erc20(struct signing_params *params,
...
+      *op = ETH_STAKING_STAKE;
+      return true;
+    }
+    if (memcmp(params->data_initial_chunk_bytes, SC_FUNC_SIG_UNSTAKE,
+               SC_FUNC_SIG_BYTES) == 0) {
+      *op = ETH_STAKING_UNSTAKE;
+      return true;
+    }
+  }
+  bool is_address_acconting =
```
[the remark this answers is not in the document]
Comment (romanz) [reply legacy/firmware/ethereum.c]: Thanks! Fixed in https://github.com/trezor/trezor-firmware/pull/4930/commits/528b899739f2229d00d24f7072977ba597925c74

File: legacy/firmware/ethereum.c [hunk 4]
```diff
@@ -567,8 +560,147 @@ static void ethereum_signing_handle_erc20(struct signing_params *params,
...
+  return false;
+}
+
+static bool layoutEthereumConfirmStakingTx(const struct signing_params *params,
+                                           enum staking_operation_t op) {
+  uint32_t args_size = params->data_initial_chunk_size - SC_FUNC_SIG_BYTES;
+  const uint8_t *args_bytes =
+      params->data_initial_chunk_bytes + SC_FUNC_SIG_BYTES;
+
+  bignum256 value, source;
```
[the remark this answers is not in the document]
Comment (romanz) [reply legacy/firmware/ethereum.c]: https://github.com/trezor/trezor-firmware/pull/4930/commits/528b899739f2229d00d24f7072977ba597925c74

**Comment by romanz** [comment]
Thanks! Pushed 2 fixups above.

**Review by mmilata — APPROVED** [review]
LGTM, not sure if you want to take a look too @matejcik. Not tested.

**Comment by romanz** [comment]
Ran ETH tests on a T1B1 device. Also tested via CLI, e.g.:
```bash
$ trezorctl eth sign-tx \
  0xD523794C879D9eC028960a231F866758e405bE34 100000000000000000 \
  -n "m/44'/60'/0'/0/0" --gas-price 20 --gas-limit 20 --nonce 0 \
  --data 3a29dbae0000000000000000000000000000000000000000000000000000000000000001
```

**Comment by romanz** [comment]
Squashed and rebased.

**Comment by romanz** [comment]
Ran ETH tests on a T1B1 device:
```
$ pytest -v 'tests/device_tests/ethereum/'
<snip>
=== 157 passed, 26 skipped in 903.78s (0:15:03) ===

```

**Comment by Ondra-Zik-SL** [comment]
**QA OK**

**Info:**
 - Suite version: web 25.5.0 (5cecbac437c6e9d75afb7e621896c7818c7d4530)
 - Browser: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36
 - OS: Linux x86_64
 - Screen: 2560x1440
 - Device: Trezor T1B1 1.13.1 regular (revision 41b786353e09390b64b107e64e1cfe3020514ad5)
 - Transport: WebUsbTransport
